doom-modeline-core.el

```diff
@@ -966,6 +975,11 @@ Example:
     \\='(bar matches \" \" buffer-info)
     \\='(media-info major-mode))
   (doom-modeline-set-modeline \\='minimal t)"
+  (when (and doom-modeline--date-not-removed-p
```

Comment:
It's not a good idea to handle date segment in `doom-modeline-def-modeline`, which is  a core function.

Regarding to overriding original string in `global-mode-string`, please refer to https://github.com/seagle0128/doom-modeline/blob/master/doom-modeline-segments.el#L2747.

Reply:
updated